diff --git a/pom.xml b/pom.xml
index 9dcb3cf..f4dcaa8 100644
--- a/pom.xml
+++ b/pom.xml
@@ -16,6 +16,7 @@
         <project.build.outputEncoding>${projectEncoding}</project.build.outputEncoding>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.deploy.skip>true</maven.deploy.skip>
     </properties>
 
     <repositories>
@@ -30,6 +31,12 @@
     </repositories>
 
     <dependencies>
+        <dependency>
+            <groupId>com.birthdates</groupId>
+            <artifactId>protocol</artifactId>
+            <version>1.0.4</version>
+            <scope>provided</scope>
+        </dependency>
         <!-- Lombok -->
         <dependency>
             <groupId>org.projectlombok</groupId>
@@ -54,17 +61,4 @@
             <scope>provided</scope>
         </dependency>
     </dependencies>
-
-    <build>
-        <resources>
-            <resource>
-                <targetPath>.</targetPath>
-                <filtering>true</filtering>
-                <directory>src/main/resources/</directory>
-                <includes>
-                    <include>*</include>
-                </includes>
-            </resource>
-        </resources>
-    </build>
 </project>
\ No newline at end of file
diff --git a/src/main/java/de/gerrygames/viarewind/legacysupport/BukkitPlugin.java b/src/main/java/de/gerrygames/viarewind/legacysupport/BukkitPlugin.java
index e34ca90..cfd6fba 100644
--- a/src/main/java/de/gerrygames/viarewind/legacysupport/BukkitPlugin.java
+++ b/src/main/java/de/gerrygames/viarewind/legacysupport/BukkitPlugin.java
@@ -10,11 +10,14 @@ import de.gerrygames.viarewind.legacysupport.listener.SoundListener;
 import de.gerrygames.viarewind.legacysupport.versioninfo.VersionInformer;
 import org.bukkit.Bukkit;
 import org.bukkit.configuration.file.FileConfiguration;
-import org.bukkit.plugin.java.JavaPlugin;
 import org.bukkit.scheduler.BukkitRunnable;
 import us.myles.ViaVersion.api.Via;
+import com.birthdates.protocol.ProtocolPlugin;
+import com.birthdates.protocol.UsePlugin;
+import us.myles.ViaVersion.api.protocol.ProtocolRegistry;
 
-public class BukkitPlugin extends JavaPlugin {
+@UsePlugin(name = "ViaRewind-Legacy-Support", weight = 0.9F)
+public class BukkitPlugin extends ProtocolPlugin {
 	private static BukkitPlugin instance;
 
 	@Override
@@ -31,25 +34,25 @@ public class BukkitPlugin extends JavaPlugin {
 				if (serverProtocol == -1) return;
 				cancel();
 				if (serverProtocol > 5 && config.getBoolean("enchanting-gui-fix"))
-					Bukkit.getPluginManager().registerEvents(new EnchantingListener(), BukkitPlugin.this);
+					Bukkit.getPluginManager().registerEvents(new EnchantingListener(), BukkitPlugin.this.getPlugin());
 				if (serverProtocol > 78 && config.getBoolean("brewing-stand-gui-fix"))
-					Bukkit.getPluginManager().registerEvents(new BrewingListener(), BukkitPlugin.this);
+					Bukkit.getPluginManager().registerEvents(new BrewingListener(), BukkitPlugin.this.getPlugin());
 				if (serverProtocol > 84 && config.getBoolean("lily-pad-fix"))
 					BoundingBoxFixer.fixLilyPad();
 				if (serverProtocol > 48 && config.getBoolean("ladder-fix"))
 					BoundingBoxFixer.fixLadder();
 				if (serverProtocol > 47 && config.getBoolean("sound-fix"))
-					Bukkit.getPluginManager().registerEvents(new SoundListener(), BukkitPlugin.this);
+					Bukkit.getPluginManager().registerEvents(new SoundListener(), BukkitPlugin.this.getPlugin());
 				if (serverProtocol > 5 && config.getBoolean("slime-fix"))
-					Bukkit.getPluginManager().registerEvents(new BounceListener(), BukkitPlugin.this);
+					Bukkit.getPluginManager().registerEvents(new BounceListener(), BukkitPlugin.this.getPlugin());
 				if (serverProtocol > 76 && config.getBoolean("elytra-fix"))
-					Bukkit.getPluginManager().registerEvents(new ElytraListener(), BukkitPlugin.this);
+					Bukkit.getPluginManager().registerEvents(new ElytraListener(), BukkitPlugin.this.getPlugin());
 				if (serverProtocol > 54 && config.getBoolean("area-effect-cloud-particles"))
-					Bukkit.getPluginManager().registerEvents(new AreaEffectCloudListener(), BukkitPlugin.this);
+					Bukkit.getPluginManager().registerEvents(new AreaEffectCloudListener(), BukkitPlugin.this.getPlugin());
 				if (config.getBoolean("versioninfo.active"))
 					new VersionInformer();
 			}
-		}.runTaskTimer(this, 1L, 1L);
+		}.runTaskTimer(getPlugin(), 1L, 1L);
 	}
 
 	public static BukkitPlugin getInstance() {
diff --git a/src/main/java/de/gerrygames/viarewind/legacysupport/listener/AreaEffectCloudListener.java b/src/main/java/de/gerrygames/viarewind/legacysupport/listener/AreaEffectCloudListener.java
index 63fc9ed..7144d60 100644
--- a/src/main/java/de/gerrygames/viarewind/legacysupport/listener/AreaEffectCloudListener.java
+++ b/src/main/java/de/gerrygames/viarewind/legacysupport/listener/AreaEffectCloudListener.java
@@ -20,7 +20,7 @@ public class AreaEffectCloudListener implements Listener {
 	private ArrayList<AreaEffectCloud> effectClouds = new ArrayList<>();
 
 	public AreaEffectCloudListener() {
-		Bukkit.getScheduler().runTaskTimer(BukkitPlugin.getInstance(), () -> {
+		Bukkit.getScheduler().runTaskTimer(BukkitPlugin.getInstance().getPlugin(), () -> {
 			Set<Player> players = Bukkit.getOnlinePlayers()
 					                      .stream()
 					                      .filter(p -> Via.getAPI().getPlayerVersion(p) <= 54)
diff --git a/src/main/java/de/gerrygames/viarewind/legacysupport/listener/SoundListener.java b/src/main/java/de/gerrygames/viarewind/legacysupport/listener/SoundListener.java
index 2999b0d..098d866 100644
--- a/src/main/java/de/gerrygames/viarewind/legacysupport/listener/SoundListener.java
+++ b/src/main/java/de/gerrygames/viarewind/legacysupport/listener/SoundListener.java
@@ -19,6 +19,7 @@ import org.bukkit.event.player.PlayerPickupItemEvent;
 import us.myles.ViaVersion.api.Via;
 
 import java.lang.reflect.Method;
+import java.util.Arrays;
 
 public class SoundListener implements Listener {
 	private static boolean isSoundCategory = false;
@@ -41,7 +42,7 @@ public class SoundListener implements Listener {
 					SoundListener.this.onItemPickUp((Player) e.getEntity());
 				}
 
-			}, BukkitPlugin.getInstance());
+			}, BukkitPlugin.getInstance().getPlugin());
 		} catch (Exception ex) {
 			Bukkit.getPluginManager().registerEvents(new Listener() {
 
@@ -50,7 +51,7 @@ public class SoundListener implements Listener {
 					SoundListener.this.onItemPickUp(e.getPlayer());
 				}
 
-			}, BukkitPlugin.getInstance());
+			}, BukkitPlugin.getInstance().getPlugin());
 		}
 	}
 
diff --git a/src/main/java/de/gerrygames/viarewind/legacysupport/versioninfo/VersionInformer.java b/src/main/java/de/gerrygames/viarewind/legacysupport/versioninfo/VersionInformer.java
index 150fb36..260ca07 100644
--- a/src/main/java/de/gerrygames/viarewind/legacysupport/versioninfo/VersionInformer.java
+++ b/src/main/java/de/gerrygames/viarewind/legacysupport/versioninfo/VersionInformer.java
@@ -20,10 +20,10 @@ public class VersionInformer implements Listener {
 		maxVersion = BukkitPlugin.getInstance().getConfig().getInt("versioninfo.max-version");
 		String interval = BukkitPlugin.getInstance().getConfig().getString("versioninfo.interval");
 		if (interval.equalsIgnoreCase("JOIN")) {
-			Bukkit.getPluginManager().registerEvents(this, BukkitPlugin.getInstance());
+			Bukkit.getPluginManager().registerEvents(this, BukkitPlugin.getInstance().getPlugin());
 		} else {
 			long ticks = Long.parseLong(interval);
-			Bukkit.getScheduler().runTaskTimer(BukkitPlugin.getInstance(), () -> {
+			Bukkit.getScheduler().runTaskTimer(BukkitPlugin.getInstance().getPlugin(), () -> {
 				Bukkit.getOnlinePlayers().forEach(player -> {
 					int version = Via.getAPI().getPlayerVersion(player);
 					if (version>maxVersion) return;
